src/lib/passes/instancegraph.cpp

```diff
@@ -0,0 +1,129 @@
+#include "coreir-pass/instancegraph.h"
+
+using namespace CoreIR;
+
+void InstanceGraph::clear() {
+  nodeMap.clear();
+  for (auto ign : sortedNodes) delete ign;
+  sortedNodes.clear();
+}
+
+void InstanceGraph::sortVisit(InstanceGraphNode* node) {
+  if (node->mark==2) return;
+  ASSERT(node->mark!=1,"SOMEHOW not a DAG");
+  node->mark = 1;
+  for (auto nextnode : node->ignList) {
+    cout << node->getInstantiable()->getName() << "->" << nextnode->getInstantiable()->getName() << endl;
```

Comment:
Should eventually add a `debug_print` macro or something that lets you silence these types of messages by default.